cmd/bosun/conf/conf.go

```diff
@@ -1,120 +1,120 @@
 package conf // import "bosun.org/cmd/bosun/conf"
 
 import (
-	"encoding/json"
+	"bytes"
 	"fmt"
-	htemplate "html/template"
-	"io/ioutil"
-	"net"
-	"net/http"
+	"hash/fnv"
 	"net/mail"
 	"net/url"
-	"os"
+	"os/exec"
 	"regexp"
-	"runtime"
-	"strconv"
 	"strings"
-	ttemplate "text/template"
 	"time"
 
-	"bosun.org/cmd/bosun/conf/parse"
+	"github.com/influxdata/influxdb/client"
+
 	"bosun.org/cmd/bosun/expr"
-	eparse "bosun.org/cmd/bosun/expr/parse"
+	"bosun.org/cmd/bosun/expr/parse"
 	"bosun.org/graphite"
 	"bosun.org/models"
 	"bosun.org/opentsdb"
+
+	htemplate "html/template"
+	ttemplate "text/template"
+
 	"bosun.org/slog"
-	"github.com/MiniProfiler/go/miniprofiler"
-	"github.com/influxdata/influxdb/client"
 )
 
-type Conf struct {
-	Vars
-	Name            string        // Config file name
-	CheckFrequency  time.Duration // Time between alert checks: 5m
-	DefaultRunEvery int           // Default number of check intervals to run each alert: 1
-	HTTPListen      string        // Web server listen address: :80
-	Hostname        string
-	RelayListen     string // OpenTSDB relay listen address: :4242
-	SMTPHost        string // SMTP address: ny-mail:25
-	SMTPUsername    string // SMTP username
-	SMTPPassword    string // SMTP password
-	Ping            bool
-	PingDuration    time.Duration // Duration from now to stop pinging hosts based on time since the host tag was touched
-	EmailFrom       string
-	StateFile       string
-	LedisDir        string
-	LedisBindAddr   string
+type SystemConfProvider interface {
+	GetHTTPListen() string
+	GetRelayListen() string
+
+	GetSMTPHost() string
+	GetSMTPUsername() string // SMTP username
+	GetSMTPPassword() string // SMTP password
+	GetPing() bool
+	GetPingDuration() time.Duration
+	GetEmailFrom() string
+	GetLedisDir() string
+	GetLedisBindAddr() string
+	GetRedisHost() string
+	GetRedisDb() int
+	GetRedisPassword() string
+	GetTimeAndDate() []int
+	GetSearchSince() opentsdb.Duration
+
+	GetCheckFrequency() time.Duration
+	GetDefaultRunEvery() int
+	GetUnknownThreshold() int
+	GetMinGroupSize() int
+
+	GetShortURLKey() string
+	GetInternetProxy() string
+
+	GetRuleFilePath() string
+	GetCommandHookPath() string
+
+	SetTSDBHost(tsdbHost string)
+	GetTSDBHost() string
+
+	GetLogstashElasticHosts() expr.LogstashElasticHosts
+	GetAnnotateElasticHosts() expr.ElasticHosts
+	GetAnnotateIndex() string
+
+	// Contexts , not sure these should be in conf but leaving them there for now
+	GetTSDBContext() opentsdb.Context
+	GetGraphiteContext() graphite.Context
+	GetInfluxContext() client.Config
+	GetLogstashContext() expr.LogstashElasticHosts
+	GetElasticContext() expr.ElasticHosts
+	AnnotateEnabled() bool
+
+	MakeLink(string, *url.Values) string
+	EnabledBackends() EnabledBackends
+}
+
+func ValidateSystemConf(sc SystemConfProvider) error {
+	if sc.GetSMTPHost() == "" && sc.GetEmailFrom() == "" {
```

Comment:
I'm not sure about this check. What if someone is only using post notifications?